src/ZoneServer/Skills/Handlers/Sapper/LegHoldTrap.cs

```diff
@@ -0,0 +1,212 @@
+		/// <param name="caster"></param>
+		/// <param name="skill"></param>
+		/// <param name="trap"></param>
+		/// <param name="cancellationToken"></param>
+		private async void AlertRange(ICombatEntity caster, Skill skill, Mob trap, Position farPos, CancellationToken cancellationToken)
+		{
+			var splashArea = new Circle(farPos, 40);
+
+			Debug.ShowShape(caster.Map, splashArea, edgePoints: false);
+			
+			while (!cancellationToken.IsCancellationRequested)
+			{
+				var targets = caster.Map.GetAttackableEntitiesIn(caster, splashArea);
+				if (targets.Count > 0)
+				{
+					await this.TriggerTrap(caster, skill, trap, farPos);
+					break;
+				}
+
+				await Task.Delay(TimeSpan.FromMilliseconds(200));
+			}
+		}
+
+		/// <summary>
+		/// Explodes the trap object and does its following effects
+		/// </summary>
+		/// <param name="caster"></param>
+		/// <param name="skill"></param>
+		/// <param name="trap"></param
+		/// <param name="farPos"></param>
+		private async Task TriggerTrap(ICombatEntity caster, Skill skill, Mob trap, Position farPos)
+		{
+			var character = caster as Character;
+			var effectId = ForceId.GetNew();
+
+			Send.ZC_DEAD(trap, trap.Position);
+			Send.ZC_NORMAL.GroundEffect_59(character, "Sapper_LegHoldTrap_Pad", skill.Id, farPos, effectId, true);
+
+			caster.Map.RemoveMonster(trap);
+
+			if (character.PlacedTraps.Contains(trap))
+			{
+				character.PlacedTraps.Remove(trap);
+			}
+
+			var cancellationTokenSource = new CancellationTokenSource();
+
+			this.AreaOfEffect(caster, farPos, skill, cancellationTokenSource.Token);
+
+			await Task.Delay(TimeSpan.FromSeconds(15));
+
+			cancellationTokenSource.Cancel();
+
+			Send.ZC_NORMAL.GroundEffect_59(character, "Sapper_LegHoldTrap_Pad", skill.Id, farPos, effectId, false);
+		}
+
+		/// <summary>
+		/// Area of effect after the trap was triggered
+		/// </summary>
+		/// <param name="caster"></param>
+		/// <param name="centerPosition"></param>
+		/// <param name="skill"></param>
+		/// <param name="cancellationToken"></param>
+		private async void AreaOfEffect(ICombatEntity caster, Position centerPosition, Skill skill, CancellationToken cancellationToken)
+		{
+			var splashArea = new Circle(centerPosition, 80);
+
+			Debug.ShowShape(caster.Map, splashArea, edgePoints: false);
+
+			await Task.Delay(TimeSpan.FromSeconds(2));
+
+			while (!cancellationToken.IsCancellationRequested)
+			{
+				var targets = caster.Map.GetAttackableEntitiesIn(caster, splashArea);
+
+				foreach (var target in targets.LimitBySDR(caster, skill))
+				{
+					// Oficial server skills apply slow while the mob is inside the area
+					// But for now we gonna apply 10 seconds slow
+					if (!target.Components.Get<BuffComponent>().Has(BuffId.Common_Slow))
+					{
+						var duration = TimeSpan.FromSeconds(10);
+						target.StartBuff(BuffId.Common_Slow, skill.Level, 0, duration, caster);
+					}
+
+					if (!target.Components.Get<BuffComponent>().Has(BuffId.LegHoldTrap_Debuff))
```

Comment:
Is this check necessary? The buff is defined with an overbuff of 1, so it shouldn't get started more than once either way.